app/components/blacklight/document/page_header_component.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+module Blacklight
+  module Document
+    # Render the start over and prev/next displays
+    class PageHeaderComponent < Blacklight::Component
+      attr_reader :document, :blacklight_config, :search_context, :search_session
+
+      delegate :blacklight_config, to: :helpers
+
+      def initialize(document:, search_context:, search_session:)
+        super
+        @search_context = search_context
+        @search_session = search_session
+        @document = document
+      end
+
+      def render?
+        search_context.present? || search_session.present? || has_header_tools?
+      end
+
+      def applied_params_component
+        return unless blacklight_config.track_search_session.applied_params_component
+
+        blacklight_config.track_search_session.applied_params_component.new
+      end
+
+      def pagination_component
+        return unless blacklight_config.track_search_session.item_pagination_component
+
+        blacklight_config.track_search_session.item_pagination_component.new(search_context: search_context, search_session: search_session, current_document: document)
+      end
+
+      def has_header_tools?
+        header_actions.any? || show_header_tools_component
+      end
+
+      def pagination_container_classes
+        has_header_tools? ? 'col-12 col-md-6 ms-auto' : ''
+      end
+
+      def header_container_classes
+        has_header_tools? ? 'row pagination-search-widgets pb-2' : 'pagination-search-widgets'
+      end
+
+      def header_actions
+        actions = helpers.filter_partials(blacklight_config.view_config(:show).header_actions, { document: document })
+        actions.map { |_k, v| v }
+      end
+
+      def show_header_tools_component
+        blacklight_config.view_config(:show).show_header_tools_component
+      end
+
+      def default_action_component_render
+        render Blacklight::Document::ActionsComponent.new(document: document,
+                                                          tag: tag,
+                                                          classes: classes,
+                                                          link_classes: link_classes,
+                                                          actions: header_actions,
+                                                          url_opts: Blacklight::Parameters.sanitize(params.to_unsafe_h))
+      end
+
+      def tag
+        'div'
+      end
+
+      def classes
+        'd-inline-flex header-tools align-items-center col-12 col-md-6 ms-auto justify-content-md-end'
+      end
+
+      def link_classes
+        'btn btn-outline-primary ms-2'
+      end
+
+      def render_header_tools
+        return unless has_header_tools?
+
+        show_header_tools_component&.tap do |show_tools_component_class|
```

Comment:
Is this the same as `return render show_header_tools_component.new(document: document) if show_header_tools_component` ?  I think writing it that way is clearer.

Reply:
True but this is the way it is in show_tools so I figured we would match? https://github.com/projectblacklight/blacklight/blob/a8b5a3d3a2e8243c8a875d40528eae5c106d83bd/app/components/blacklight/document/sidebar_component.rb#L20-L22